analyses/upstream-analysis/01B_run_seurat_qc_multiple_samples.R

```diff
@@ -0,0 +1,125 @@
+  library(yaml)
+  library(optparse)
+})
+
+#################################################################################
+# load config file
+configFile <- paste0("../../project_parameters.Config.yaml")
+if (!file.exists(configFile)){
+  cat("\n Error: configuration file not found:", configFile)
+  stop("Exit...")}
+
+# read `yaml` file defining the `params` of the project and strategy analysis
+yaml <- read_yaml(configFile)
+
+#################################################################################
+# Set up directories and paths to file Inputs/Outputs
+root_dir <- yaml$root_dir
+metadata_dir <- yaml$metadata_dir
+analysis_dir <- file.path(root_dir, "analyses", "upstream-analysis") 
+
+# File path to plots directory
+plots_dir <-
+  file.path(analysis_dir, "plots") 
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# File path to plots directory for seurat_qc
+seurat_qc_plots_dir <-
+  file.path(plots_dir, "01_Seurat_qc") 
+if (!dir.exists(seurat_qc_plots_dir)) {
+  dir.create(seurat_qc_plots_dir)
+}
+
+# Create module_results_dir
+module_results_dir <- 
+  file.path(analysis_dir, paste0("results"))
+if (!dir.exists(module_results_dir)) {
+  dir.create(module_results_dir)
+}
+
+# Create seurat_results_dir
+seurat_results_dir <- 
+  file.path(module_results_dir, paste0("01_Seurat_qc"))
+if (!dir.exists(seurat_results_dir)) {
+  dir.create(seurat_results_dir)
+}
+
+#######################################################
+# Read metadata file and define `sample_name`
+metadata_file <- file.path(metadata_dir, "project_metadata.tsv") # metadata input file
+
+# Read metadata file and define `sample_name`
+project_metadata <- read.csv(metadata_file, sep = "\t", header = TRUE)
+sample_name <- unique(as.character(project_metadata$ID))
+sample_name <- sort(sample_name, decreasing = FALSE)
+print(sample_name)
+
+#####################################################################################
+# Run markdown script per each library
+for (i in seq_along(sample_name)){
+
+  # Create directory to save html reports
+  samples_plots_dir <- file.path(seurat_qc_plots_dir, sample_name[i]) 
+  if (!dir.exists(samples_plots_dir)) {
+    dir.create(samples_plots_dir)}
+    
+  # Create results_dir per sample
+  results_dir <- file.path(seurat_results_dir, sample_name[i])
+  if (!dir.exists(results_dir)) {
+    dir.create(results_dir)}
+  
+  # Render and save html
+  rmarkdown::render("01A_run_seurat_qc.Rmd", 
+                    output_dir = file.path(samples_plots_dir),
+                    clean = TRUE, # using TRUE will clean intermediate files that are created during rendering
+                    output_file = c(paste('Report-', 'seurat-qc', '-', sample_name[i], '-', Sys.Date(), sep = '')),
+                    output_format = 'all',
+                    params = list(use_condition_split = yaml$use_condition_split_seurat_multiple_samples,
+                                  print_pdf = yaml$print_pdf_seurat_multiple_samples,
```

Comment:
For this parameter and the parameter above, it seems like there are not corresponding parameters defined in the current yaml file. Mostly just for my own understanding, will those be added to the yaml later when we start doing the actual test runs?

Reply:
This has been addressed here 149235824da7fdab83c6a25681f1fcfb51c6b34e.